### DISCUSSION TITLE
How to redirect to a 404.tsx page in case of Page not found?

### DISCUSSION THREAD

**Discussion by anavisgueira** [mui/toolpad#4417] [Help and support] [answered]
How can I have a personal "Page not found" in case of error 404? I´ve read that I have to create a 404.tsx page in a "\pages" directory, but it doesn´t work. Tks a lot!

**Comment by bharatkashyap** [comment, marked as answer]
I'm assuming you're using the`pages` directory together with the `app` directory. This is a Next.js issue https://github.com/vercel/next.js/issues/58945, but the workaround (mentioned in the issue comments) is to use the `not-found.tsx` file in the `app` directory instead of `pages/404.tsx`

**Reply by anavisgueira** [reply]
thanks for the answer, It solved!
